cpp/src/parquet/size_statistics_test.cc

```diff
@@ -376,4 +376,22 @@ TEST_F(SizeStatisticsRoundTripTest, LargePage) {
                                                         /*byte_array_bytes=*/{90000}}));
 }
 
+TEST_F(SizeStatisticsRoundTripTest, MaxLevelZero) {
+  auto schema =
+      ::arrow::schema({::arrow::field("a", ::arrow::utf8(), /*nullable=*/false)});
+  WriteFile(SizeStatisticsLevel::PageAndColumnChunk,
+            ::arrow::TableFromJSON(schema, {R"([["foo"],["bar"]])"}),
+            /*max_row_group_length=*/2,
+            /*page_size=*/1024);
+  ReadSizeStatistics();
```

Comment:
Does it also read the full file? We should ensure we don't crash elsewhere.

Reply:
`ReadSizeStatistics` currently only reads the metadata where it crashes from the issue. Let me add a new function to verify scanning data.